js/stronghold.js

```diff
@@ -0,0 +1,459 @@
+                    ],
+                    'post': [
+                        'venues/{venueId}/accounts',
+                        'venues/{venueId}/accounts/{accountId}/orders',
+                        'venues/{venueId}/accounts/{accountId}/deposit',
+                        'venues/{venueId}/accounts/{accountId}/withdrawal',
+                        'venues/{venueId}/custody/accounts/{accountId}/payment',
+                        'venues/{venueId}/custody/accounts/{accountId}/payment/{paymentId}/stop',
+                        'venues/{venueId}/custody/accounts/{accountId}/operations/{operationId}/signatures',
+                        'venues/{venueId}/anchor/withdrawal',
+                    ],
+                },
+            },
+            'options': {
+                'venues': {
+                    'main': 'trade-public',
+                    'sandbox': 'sandbox-public',
+                },
+                'paymentMethods': {
+                    'ETH': 'ethereum',
+                    'BTC': 'bitcoin',
+                    'XLM': 'stellar',
+                },
+            },
+            'exceptions': {
+                'CREDENTIAL_MISSING': AuthenticationError,
+                'CREDENTIAL_INVALID': AuthenticationError,
+                'CREDENTIAL_REVOKED': AccountSuspended,
+                'CREDENTIAL_NO_IDENTITY': AuthenticationError,
+                'PASSPHRASE_INVALID': AuthenticationError,
+                'SIGNATURE_INVALID': AuthenticationError,
+                'TIME_INVALID': InvalidNonce,
+                'BYPASS_INVALID': AuthenticationError,
+                'INSUFFICIENT_FUNDS': InsufficientFunds,
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetVenuesVenueIdMarkets (params);
+        const data = response['result'];
+        // [ { id: 'SHXUSD',
+        //     baseAssetId: 'SHX/stronghold.co',
+        //     counterAssetId: 'USD/stronghold.co',
+        //     minimumOrderSize: '1.0000000',
+        //     minimumOrderIncrement: '1.0000000',
+        //     minimumPriceIncrement: '0.00010000',
+        //     displayDecimalsPrice: 4,
+        //     displayDecimalsAmount: 0 }, ... ]
+        let result = {};
+        for (let i = 0; i < data.length; i++) {
+            const entry = data[i];
+            const marketId = entry['id'];
+            const baseId = entry['baseAssetId'];
+            const quoteId = entry['counterAssetId'];
+            const baseAssetId = baseId.split ('/')[0];
+            const quoteAssetId = quoteId.split ('/')[0];
+            const base = this.commonCurrencyCode (baseAssetId);
+            const quote = this.commonCurrencyCode (quoteAssetId);
+            const symbol = base + '/' + quote;
+            const limits = {
+                'amount': {
+                    'min': this.safeFloat (entry, 'minimumOrderSize'),
+                    'max': undefined,
+                },
```

Comment:
We should always populate all fields in the structure, even the undefined ones. For limits in this particular case we can set min price to their `minimumPriceIncrement`, max price to `undefined`. The cost limits would be both undefined, but still should be in the structure – the reason behind this is that adding intermediary checks for presence of keys everywhere is less convenient from the user perspective.